hibernate-core/src/test/java/org/hibernate/orm/test/fetch/FetchModeSubselectTest.java

```diff
@@ -0,0 +1,266 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or http://www.gnu.org/licenses/lgpl-2.1.html
+ */
+package org.hibernate.orm.test.fetch;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.hibernate.annotations.Fetch;
+import org.hibernate.annotations.FetchMode;
+
+import org.hibernate.testing.TestForIssue;
+import org.hibernate.testing.orm.junit.DomainModel;
+import org.hibernate.testing.orm.junit.SessionFactory;
+import org.hibernate.testing.orm.junit.SessionFactoryScope;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import jakarta.persistence.CascadeType;
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.FetchType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.JoinTable;
+import jakarta.persistence.ManyToMany;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.OneToMany;
+import jakarta.persistence.Table;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+@DomainModel(
+		annotatedClasses = {
+				FetchModeSubselectTest.User.class,
+				FetchModeSubselectTest.Role.class,
+				FetchModeSubselectTest.Agency.class
+		}
+)
+@SessionFactory
+@TestForIssue(jiraKey = "HHH-15453")
+public class FetchModeSubselectTest {
+
+	@BeforeEach
+	public void setUp(SessionFactoryScope scope) {
+		scope.inTransaction(
+				session -> {
+					User ann = new User( 1, "Ann" );
+					User bob = new User( 2, "Bob" );
+
+					Role manager = new Role( 1, "manager" );
+					Role developer = new Role( 2, "developer" );
+
+					ann.addRole( manager );
+
+					Agency agency = new Agency( 100, "Test Agency" );
+
+					agency.addUser( ann );
+					agency.addUser( bob );
+
+					session.persist( ann );
+					session.persist( bob );
+					session.persist( manager );
+					session.persist( developer );
+					session.persist( agency );
+				}
+		);
+	}
+
+	@AfterEach
+	public void tearDown(SessionFactoryScope scope){
+		scope.inTransaction(
+			session -> {
+				session.createMutationQuery( "delete from User" ).executeUpdate();
+				session.createMutationQuery( "delete from Role" ).executeUpdate();
+				session.createMutationQuery( "delete from Agency" ).executeUpdate();
+			}
+		);
+	}
+
+	@Test
+	public void testSelectUserWithRole(SessionFactoryScope scope) {
+		scope.inSession(
+				session -> {
+					List<User> userList = session.createQuery( "FROM User WHERE id = 2", User.class ).list();
+					assertThat( userList.size() ).isEqualTo( 1 );
+					assertThat( userList.get( 0 ).getRoles().size() ).isEqualTo( 0 );
```

Comment:
Looks like this should rather be:

```suggestion
					List<User> userList = session.createQuery( "FROM User WHERE id = 1", User.class ).list();
					assertThat( userList.size() ).isEqualTo( 1 );
					assertThat( userList.get( 0 ).getRoles().size() ).isEqualTo( 1 );
```

Reply:
:+1: